### DISCUSSION TITLE
How to connect two TM1637 displays

### DISCUSSION THREAD

**Discussion by ronnieSVK** [arendst/Tasmota#20240] [Q&A]
Dear community !

i would like to kindly ask you for workaround, how to connect two identical TM1637 displays on one ESP32 running "Tasmota Display". My usecase would be displaying temperatures from two sensors (located in different places in/out) in two separate places. I managed to run one TM1637. The second display starts after boot, but no switch appeared in UI. The two displays was wired to different GPiOs and was configured in UI.

Thank you very much for your assistance !

**Comment by sfromis** [comment]
From what I can see, the current driver for TM1637 does not support multiple displays. If this was supported, you should've seen a dropdown appearing to the right when configuring the gpio component for the display, otherwise configuring the same component for multiple gpios generally does not work.

**Reply by ronnieSVK** [reply]
Thank you very much for your response ! So i have to find a different approach :(

**Reply by Noschvie** [reply]
Why not using an ESP32 for each display?

**Reply by ronnieSVK** [reply]
Thanks for you feedback. So, i can, but... The device will be used in the garage. Relay as a door opener, magnetic contact sensor for door, two temperature sensors for inside temperature (to control inside temperature above freeze level, in case of some extraordinary cold weather) and outside temperature for checking, having this information also visible in front of eyes and in MQTT. My goal was to be able to display both temperatures in garage on mentioned displays. In order to have some easy, long therm and cheap solution i have choosen the TM1637 displays. Since this is my first case to "deal with Tasmota" differently as a LED controller or FW in Power Plug, it was unknown for me, that i cant use two separate displays :/ Actually ESPhome should "know" this, but it doesnt fit into my integration.

**Reply by Noschvie** [reply]
Have a look to this [2.9inch E-Paper Display](https://www.amazon.de/dp/B07Q5PZQ9F?_encoding=UTF8&ref_=cm_sw_r_cp_ud_dp_JCBCQ9K3A8QC2PJ49FSD&th=1)

**Reply by sfromis** [reply]
If you want bright 7-segment leds, Tasmota does support up to 8 HT16K33 displays via I2C. https://github.com/arendst/Tasmota/discussions/20202

Another Tasmota approach, if you want to use the hardware you already have, could be to expand the existing driver, or code a driver in Berry (not as daunting as it may sound).

What I'm currently using for displays is ILI9341 (320 x 240 TFT) display shields with a ESP32 module on the back.

**Comment by lalo-uy** [comment]
With the same driver you can use TM1638 or MAX7219 with 8 digit that allow
showing both temps.

https://tasmota.github.io/docs/TM163x/
